### ISSUE TITLE
Can use Rector to remove property types?

### ISSUE THREAD

**Issue by leoloso** [rectorphp/rector#4051]
I've read that Rector is meant to upgrade code from a lower to a higher PHP version. Could it possibly work in the opposite direction too?

To be specific: my code needs to support PHP 7.2. But I'd like to introduce PHP 7.4's typed properties (so I can analyze it with PHPStan as to avoid bugs from conflicting types), and then use Rector to remove the property types when creating the installable release.

In summary, can I do this?:

- Write property types: `protected string $someVar;` (PHP 7.4)
- Run Rector to remove types: `protected $someVar;` (PHP 7.2)
- Save output as release

**Comment by dobryy** [comment]
Hi @leoloso,

This rule can help you https://github.com/rectorphp/rector/blob/master/docs/rector_rules_overview.md#downgradetypedpropertyrector

**Comment by leoloso** [comment]
This is exactly what I need, thanks @dobryy 

Same case concerning PHP 8, do you know of any rule to remove the type unions?

For instance, to replace: `function foo(string|int $bar): float|bool` with `function foo($bar)`

**Comment by dobryy** [comment]
@leoloso there is no rule for union types at the moment but there is a plan to extend Downgrade set of rules.

**Comment by leoloso** [comment]
Magnificent, I'll be checking on that, thanks @dobryy!

**Comment by TomasVotruba** [comment]
Closing as answered
